device/common/include/traccc/clusterization/device/impl/aggregate_cluster.ipp

```diff
@@ -140,6 +151,21 @@ inline void aggregate_cluster(
     out.measurement_id = link;
     // Set the dimensionality of the measurement.
     out.meas_dim = module_descr.dimensions();
+
+    scalar delta0 = max_channel0 - min_channel0;
+    scalar delta1 = max_channel1 - min_channel1;
+
+    if (cfg.diameter_strategy == clustering_diameter_strategy::CHANNEL0) {
+        out.diameter = delta0;
+    } else if (cfg.diameter_strategy ==
+               clustering_diameter_strategy::CHANNEL1) {
+        out.diameter = delta1;
+    } else if (cfg.diameter_strategy == clustering_diameter_strategy::MAXIMUM) {
+        out.diameter = std::max(delta0, delta1);
+    } else if (cfg.diameter_strategy ==
+               clustering_diameter_strategy::DIAGONAL) {
+        out.diameter = math::sqrt(delta0 * delta0 + delta1 * delta1);
+    }
```

Comment:
A `switch` expression may be a bit easier for the compiler to optimize here. 🤔 (I like `switch` statements...)

Reply:
Hmm, why? I find the if-else construct a lot easier to read and there is definitely no performance difference.